### ISSUE TITLE
Implementing `__getattr__`  does not make objects subscriptable

### ISSUE THREAD

**Issue by not-my-profile** [microsoft/pyright#3109]
```python
from typing import Any

class Foo:
    def __getattr__(self, s: str) -> Any:
        raise NotImplementedError()

Foo()['test']
```

Mypy and pytype correctly report that the last line is an error (Value of type "Foo" is not indexable), Pyright does not detect the error.

**Comment by JelleZijlstra** [comment]
As background, access to special methods like `__getitem__` bypasses `__getattr__` and `__getattribute__`. This is documented (not very precisely) at https://docs.python.org/3.10/reference/datamodel.html#special-method-lookup. It should apply to all methods backed by C-level "slots", as listed in https://docs.python.org/3.10/c-api/typeobj.html#quick-reference.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.225, which I just published. It will also be included in the next release of pylance.
